lib/smart_proxy_dynflow_core/logger_middleware.rb

```diff
@@ -0,0 +1,28 @@
+module SmartProxyDynflowCore
+  class LoggerMiddleware
+    def initialize(app)
+      @logger = SmartProxyDynflowCore::Log.instance
+      @app = app
+    end
+
+    def call(env)
+      status = 500
+      env['rack.logger'] = @logger
+      @logger.info { "Started #{env['REQUEST_METHOD']} #{env['PATH_INFO']} #{env['QUERY_STRING']}" }
+      @logger.debug { 'Headers: ' + env.select {|k,v| k.start_with? 'HTTP_'}.inspect }
+      @logger.debug { (body = env['rack.input'].read).empty? ? '' : 'Body: ' + body }
+      env['rack.input'].try(:seek, 0) if @logger.debug?
```

Comment:
would be better to put this inot the block above + I would prefer `rewind` instead